app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterActivity.cs

```diff
@@ -0,0 +1,124 @@
+                }
+            };
+            passwordInput.TextChanged += (sender, e) => {
+                if (string.IsNullOrEmpty((sender as EditText).Text))
+                {
+                    usernameInput.SetError(GetString(Resource.String.password_empty_error), null);
+                }
+            };
+
+            usernameInput.KeyPress += (sender, e) => {
+                if (e.KeyCode == Android.Views.Keycode.Enter && e.Event.Action == Android.Views.KeyEventActions.Down)
+                {
+                    e.Handled = true;
+                    if (!string.IsNullOrEmpty((sender as EditText).Text))
+                    {
+                        this.HideKeyboard(usernameInput);
+                        passwordInput.RequestFocus();
+                        this.ShowKeyboard(passwordInput);
+                    }
+                    else
+                    {
+                        usernameInput.SetError(GetString(Resource.String.username_empty_error), null);
+                    }
+                }
+                else
+                {
+                    e.Handled = false;
+                }
+            };
+            passwordInput.KeyPress += (sender, e) => {
+                if (e.KeyCode == Android.Views.Keycode.Enter && e.Event.Action == Android.Views.KeyEventActions.Down)
+                {
+                    e.Handled = true;
+                    if (string.IsNullOrEmpty((sender as EditText).Text))
+                    {
+                        passwordInput.SetError(GetString(Resource.String.password_empty_error), null);
+                    }
+                    else
+                    {
+                        this.HideKeyboard(passwordInput);
+                    }
+                }
+                else
+                {
+                    e.Handled = false;
+                }
+            };
+
+            registerButton.Click += async(sender, e) =>
+            {
+                if(string.IsNullOrEmpty(usernameInput.Text))
+                {
+                    usernameInput.SetError(GetString(Resource.String.username_empty_error), null);
+                }
+                else if(string.IsNullOrEmpty(passwordInput.Text))
+                {
+                    passwordInput.SetError(GetString(Resource.String.username_empty_error), null);
+                }
+                else
+                {
+                    var regex = new Regex(@"^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[#$^+=!*()@%&]).{8,15}$");
+                    if(regex.IsMatch(passwordInput.Text))
+                    {
+                        var response = await registerViewModel.AddUser(usernameInput.Text, passwordInput.Text);
+                        if(response == Response.OK)
+                        {
+                            Toast.MakeText(this, Resource.String.user_created, ToastLength.Short).Show();
+                            StartActivity(new Android.Content.Intent(this, typeof(SignInActivity)));
```

Comment:
Don't finish sign in activity, when register button(at sign in activity) is clicked. And here just finish the register activity without starting new sign in activity.

Reply:
And you can also set the result as OK, and then OnActivityResult in SignInActivity you could show Toast message of successful registration